opm/simulators/aquifers/AquiferInterface.hpp

```diff
@@ -233,7 +233,99 @@ class AquiferInterface
+
+        // Translate the C face tag into the enum used by opm-parser's TransMult class
+        Opm::FaceDir::DirEnum faceDirection;
+
+        // denom_face_areas is the sum of the areas connected to an aquifer
+        Scalar denom_face_areas = 0.;
+        this->cellToConnectionIdx_.resize(this->ebos_simulator_.gridView().size(/*codim=*/0), -1);
+        const auto& gridView = this->ebos_simulator_.vanguard().gridView();
+        for (size_t idx = 0; idx < this->size(); ++idx) {
+            const auto global_index = this->connections_[idx].global_index;
+            const int cell_index = this->ebos_simulator_.vanguard().compressedIndex(global_index);
+            auto elemIt = gridView.template begin</*codim=*/ 0>();
+            if (cell_index > 0)
+                std::advance(elemIt, cell_index);
+
+           //the global_index is not part of this grid
+            if ( cell_index < 0 || elemIt->partitionType() != Dune::InteriorEntity)
+                continue;
+
+            this->cellToConnectionIdx_[cell_index] = idx;
+            this->cell_depth_.at(idx) = this->ebos_simulator_.vanguard().cellCenterDepth(cell_index);
+        }
+        // get areas for all connections
+        ElementMapper elemMapper(gridView, Dune::mcmgElementLayout());
+        auto elemIt = gridView.template begin</*codim=*/ 0>();
+        const auto& elemEndIt = gridView.template end</*codim=*/ 0>();
+        for (; elemIt != elemEndIt; ++elemIt) {
+            const auto& elem = *elemIt;
+            unsigned cell_index = elemMapper.index(elem);
+            int idx = this->cellToConnectionIdx_[cell_index];
+
+            // only deal with connections given by the aquifer
+            if( idx < 0)
+                continue;
+
+            auto isIt = gridView.ibegin(elem);
+            const auto& isEndIt = gridView.iend(elem);
+            for (; isIt != isEndIt; ++ isIt) {
+                // store intersection, this might be costly
+                const auto& intersection = *isIt;
+
+                // only deal with grid boundaries
+                if (!intersection.boundary())
+                    continue;
+
+                int insideFaceIdx  = intersection.indexInInside();
+                switch (insideFaceIdx) {
+                case 0:
+                    faceDirection = Opm::FaceDir::XMinus;
+                    break;
+                case 1:
+                    faceDirection = Opm::FaceDir::XPlus;
+                    break;
+                case 2:
+                    faceDirection = Opm::FaceDir::YMinus;
+                    break;
+                case 3:
+                    faceDirection = Opm::FaceDir::YPlus;
+                    break;
+                case 4:
+                    faceDirection = Opm::FaceDir::ZMinus;
+                    break;
+                case 5:
+                    faceDirection = Opm::FaceDir::ZPlus;
+                    break;
+                default:
+                    OPM_THROW(Opm::NumericalIssue,
```

Comment:
This should have been a logic_error all along i believe, and the error message is misleading. Maybe just "Internal error in aquifer setup." or similar.